test/unit/range/view/view_translate_join_test.cpp

```diff
@@ -0,0 +1,158 @@
+
+    // user-defined frame combination
+    auto v7 = vec | view::translate_join(translation_frames::FWD_FRAME_0 | translation_frames::FWD_FRAME_2);
+    // == [[T,Y,V,R],[V,R,T],[S,R,A,L],[E,S,F,S]]
+    EXPECT_EQ(v7.size(), cmp5.size());
+    for (unsigned i = 0; i < v7.size(); i++)
+        EXPECT_TRUE((std::ranges::equal(v7[i], cmp5[i])));
+
+    // function syntax
+    auto v8 = view::translate_join(vec, translation_frames::FWD_REV_0);
+    // == [[T,Y,V,R],[R,T,Y,V],[V,R,T],[Y,V,R,T],[T,Y,V,R],[R,T,Y],[S,R,A,L], [R,E,L,*], [E,S,F,S], [A,K,A,L], [L,K,L,S], [*,S,S,R]]
+    EXPECT_EQ(v8.size(), cmp2.size());
+    for (unsigned i = 0; i < v8.size(); i++)
+        EXPECT_TRUE((std::ranges::equal(v8[i], cmp2[i])));
+
+    // combinability
+    auto v9 = vec | view::complement | view::translate_join(translation_frames::FWD_REV_0);
+    // == [[C,M,H,A],[M,H,A,C],[S,S,R,N],[R,F,R,E]]
+    EXPECT_EQ(v9.size(), cmp6.size());
+    for (unsigned i = 0; i < v9.size(); i++)
+        EXPECT_TRUE((std::ranges::equal(v9[i], cmp6[i])));
+
+    // combinability
+    auto v10 = vec | view::complement | view::translate_join(translation_frames::FWD_REV_0) | std::view::take(1);
+    // == [C,M,H,A]
+    EXPECT_EQ(v10.size(), cmp7.size());
+    for (unsigned i = 0; i < v10.size(); i++)
+        EXPECT_TRUE((std::ranges::equal(v10[i], cmp7[i])));
+
+    // combinability and function syntax
+    auto v11 = detail::view_translate_join(view::complement(vec), translation_frames::FWD_REV_0);
+    // == [[C,M,H,A],[M,H,A,C],[S,S,R,N],[R,F,R,E]]
+    EXPECT_EQ(v11.size(), cmp6.size());
+    for (unsigned i = 0; i < v11.size(); i++)
+        EXPECT_TRUE((std::ranges::equal(v11[i], cmp6[i])));
+}
+
+TYPED_TEST(nucleotide, view_translate_concepts)
+{
+    std::string const in1{"ACGTACGTACGTA"};
+    std::string const in2{"TCGAGAGCTTTAGC"};
+    std::vector<std::vector<TypeParam> > vec;
+    vec.resize(2);
+    vec[0] = in1 | view::char_to<TypeParam>;
+    vec[1] = in2 | view::char_to<TypeParam>;
+
+    EXPECT_TRUE(std::ranges::ForwardRange<decltype(vec)>);
+    EXPECT_TRUE(std::ranges::RandomAccessRange<decltype(vec)>);
+    EXPECT_TRUE(std::ranges::SizedRange<decltype(vec)>);
+
+    auto v1 = vec | view::translate_join(translation_frames::FWD_REV_0);
+
+    EXPECT_TRUE(std::ranges::InputRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::ForwardRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::RandomAccessRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::SizedRange<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::View<decltype(v1)>);
+    EXPECT_TRUE(std::ranges::RandomAccessRange<value_type_t<decltype(v1)>>);
+    EXPECT_TRUE(std::ranges::SizedRange<value_type_t<decltype(v1)>>);
+    EXPECT_TRUE(std::ranges::View<value_type_t<decltype(v1)>>);
+    EXPECT_TRUE(std::ranges::RandomAccessRange<reference_t<decltype(v1)>>);
+    EXPECT_TRUE(std::ranges::SizedRange<reference_t<decltype(v1)>>);
+    EXPECT_TRUE(std::ranges::View<reference_t<decltype(v1)>>);
+}
```

Comment:
We now have a test_template for the iterators of ranges. Can you ask @smehringer how to apply this for your view? It's likely only a few lines...